### ISSUE TITLE
Restarting (docker based deployments) corrupts something inside dgraph

### ISSUE THREAD

**Issue by cajus** [dgraph-io/dgraph#2374]
- What version of Dgraph are you using?
1.0.5 / docker

- Have you tried reproducing the issue with latest release?
Yes

- What is the hardware spec (RAM, OS)?
Small docker test instance with a couple of entries.

- Steps to reproduce the issue (command/config used to run Dgraph).
If you i.e. restart the docker image, it's very often the case that *something* is corrupted. Completely re-initializing the data store makes it work again - but that should be really more robust. Here's an example:

I'm not able to modify any entries that are inside the tree anymore. This protocol shows the problem:

**Query the entry to be modified**

```
{
  q(func: uid(0x2711)) {
    uid
    name
    description
  }
}
```

**Result**

```
{
  "data": {
    "q": [
      {
        "uid": "0x2711",
        "name": "Foobar"
      }
    ]
  }
}
```

**Mutation**

```
{
  set {
    <0x2711> <name> "Barfoo" .
  }
}
```

**Result**

`Uid: [10001] cannot be greater than lease: [0]`

- Expected behaviour and actual result.

`Done`

**Comment by MichelDiz** [comment]
Uid: [10001] cannot be greater than lease: [0] can happen when the Dgraph is empty. But in your case can be related with "Locks". The data still there, but you can't acess with locks on Dgraph files.

The rule is, you can use UIDs created by Dgraph only. So if you're getting this error with your context explaned. It's seems to me that is Locks case.

But, I did a very simple test here in my Windows 10 Machine with Docker ToolBox (VirtualBox) forcing restart and forcing turn off. But I was able to overrite the node as you explains in your case.

You still having this issue? This UID showed is given from Dgraph for sure?

Please, if you can do an equal test with the node with "0x1" and with the greater or more "different" UID that has giving problem. Just to have a more empirical notion of the case.

Try to go inside the Docker Volume and delete the "lock" file in Dgraph folders. And run the Dgraph Image again.

PS. I used Ratel for the tests.

**Comment by pawanrawal** [comment]
This problem should only happen if you clear the `zw` directory that is created by Zero. Is that the case?

**Comment by manishrjain** [comment]
Looks like the last question was unanswered. @cajus can you try without Docker and see if this is a Dgraph issue, or some Docker config issue?

**Comment by cajus** [comment]
Sorry. I've been on vacation for the last two weeks and lost the focus on that issue after returning. So - thanks for the ping. I talked with the "deployment guy", because I passed over the hint given by @pawanrawal before the vacation. It turned out that he disabled the storage for Zero, because it broke every time we scaled the service up or down.

Without the storage the reported issue happened. Re-adding the storage makes it work fine - with a constant cluster size. Changing the size, breaks it. But that's another issue I guess.
